### PR TITLE
Filtering and searching

### PR DESCRIPTION
**Pull request by Iggelds**

### CODE REVIEW

**Review by IrinaPodoliak — APPROVED** [review]
Have no comments)

**Comment by malmicha** [comment]
Approved!

File: src/test/java/FilterAndSearch.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        webDriver.manage().timeouts().implicitlyWait(Duration.ofSeconds(2));
+
+        List<WebElement> searchResults = webDriver.findElements(By.xpath("//*[@data-component-type=\"s-search-result\"]"));
+        for (WebElement element : searchResults) {
+            try {
+                WebElement whole = element.findElement(By.className("a-price-whole"));
+                WebElement fraction = element.findElement(By.className("a-price-fraction"));
+                String result = whole.getText() + "." + fraction.getText();
+                double price = Double.parseDouble(result);
+                assert price >= 100.00;
```
Comment (juliaHrabovska) [inline src/test/java/FilterAndSearch.java:68]: It is not recommended of using assert, please use testNG library methods

File: src/test/java/FilterAndSearch.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    @Test
+    public void PriceVerify() {
+        WebElement lowPrice = webDriver.findElement(By.xpath("//*[@id=\"low-price\"]"));
+        lowPrice.sendKeys("100");
+        WebElement highPrice = webDriver.findElement(By.xpath("//*[@id=\"high-price\"]"));
+        highPrice.sendKeys("150");
+        WebElement submitButton = webDriver.findElement(By.xpath("//*[@id=\"priceRefinements\"]//form"));
+        submitButton.submit();
+        webDriver.manage().timeouts().implicitlyWait(Duration.ofSeconds(2));
```
Comment (juliaHrabovska) [inline src/test/java/FilterAndSearch.java:59]: implicitlyWait is not recommended. Use explicit wait is instead

File: src/test/java/FilterAndSearch.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        for (WebElement element : searchResults) {
+            try {
+                WebElement whole = element.findElement(By.className("a-price-whole"));
+                WebElement fraction = element.findElement(By.className("a-price-fraction"));
+                String result = whole.getText() + "." + fraction.getText();
+                double currentPrice = Double.parseDouble(result);
+                assert currentPrice >= price;
+                price = currentPrice;
+            } catch (Exception e) {
+            }
```
Comment (juliaHrabovska) [inline src/test/java/FilterAndSearch.java:95]: If you use try-catch, please put some logic in a catch block. 
And it is okay that your tests failed, I know that functionality can work incorrectly, so it is better to remove this try-catch block

**Pull request merged**